lib/src/lib.rs

```diff
@@ -389,15 +387,16 @@ impl<'a> ChunkAccessor for DefaultChunkAccessor<'a> {
             data
         };
 
-        let mut sha256 = sha2::Sha256::new();
+        let mut sha256 = Sha256::default();
         sha256.input(&data);
-        let mut sha256_digest = vec![0u8; DIGEST_SIZE];
-        sha256.result(&mut sha256_digest);
-        if sha256_digest != digest {
+        let mut vec_result = vec![0u8; DIGEST_SIZE];
+        vec_result.copy_from_slice(&sha256.result());
```

Comment:
Seems to me that `Sha256::default()` (and previous `sha2::Sha256::new`) could be something ike `Sha256::default(&'a mut [u8;32]) -> Sha256<'a>`, taking space for digest from the caller. So there's no need to copy anything. I'm not sure if I understand your answer - I'm distracted ATM. :)